morse_code.rb

```diff
@@ -0,0 +1,37 @@
+def morse_hash
+  { A: '.-', B: '-...', C: '-.-.', D: '-..', E: '.', F: '..-.', G: '--.', H: '....',
+    I: '..', J: '.---', K: '-.-', L: '.-..', M: '--', N: '-.', O: '---', P: '.--.',
+    Q: '--.-', R: '.-.', S: '...', T: '-', U: '..-', V: '...-', W: '.--', X: '-..-', Y: '-.--', Z: '--..' }
+end
+
+def decode_char(string)
+  morse_hash.each do |key, value|
+    if value == string
+      letter = key.to_s
+      return letter
+    end
+  end
+end
+
+def decode_word(string)
+  split_letters = string.split
+  word = ''
+  split_letters.each do |el|
+    word += decode_char(el)
+  end
+  word
+end
+
+def decode_sentense(string)
+  new_string = string.gsub('/', ' ')
+  split_word = new_string.split('  ')
```

Comment:
- [ ] when running the below code:
```
p decode_char('.-')
p decode_word('-- -.--')
p decode("-- -.--   -. .- -- .")
p decode(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")
```
the program returns:
"A"
"MY"
"MY NAME "
"A BOX FULL  OF  RUBIES "
As you can see in the last line there are too many spaces in the returned string, this is due to the fact that in line 27 you split the words after 2 spaces, therefore if there are 4 spaces like in the test example provided by Microverse the return string will have to spaces.
If you add a space in this split it will run as expected.

Reply:
Thanks for the observation, this was a typo from us, the requirement clearly said 3 spaces and we typed 2. Thanks this is now fixed and the program works perfectly.

Reply:
![image](https://user-images.githubusercontent.com/53021807/150212052-203eee47-72ed-4a37-af80-57f7f16b91a4.png)